midend/interpreter.cpp

```diff
@@ -727,6 +727,18 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
         auto li = l->to<SymbolicInteger>();
         clone->expr = li->constant;
         DoConstantFolding cf(refMap, typeMap);
+        if (auto cast = expression->to<IR::Cast>()) {
+            if (cast->destType->is<IR::Type_Bits>()) {
```

Comment:
This might be lacking defensive programming. There is only once specific bool cast allowed as far as I recall and that is casting to bits of width 1.

Reply:
We also need to check the size of this Bit? To make it fit the size 1